### ISSUE TITLE
Not precise configuration of xgboost

### ISSUE THREAD

**Issue by sh1ng** [Microsoft/LightGBM#9]
> LightGBM uses leaf-wise algorithm instead and controls model complexity by num_leaves. So we cannot compare them in the exact same model setting. For the tradeoff, we use xgboost with max_depth=8, which will have max number leaves to 255

You should use **min_child_weight** instead it controls second sum derivative. In case of regression task it equivalent to number of nodes(divided by 2). For you objective functions it's no so trivial, but it's better that default min_child_weight=1. 

If you only set max_depth=8 it guarantys only that max number of leafes, but destitution could be really skewed and overfit train data set(in your case with default min_data_in_leaf it's less skewed ).  
Besides that increase that parameter should make you computation little faster(you no need to calculate gain if your sum less then min_child_weight).

PS
To make your benchmark more representative take a wining solution from kaggle(for instance) and try to get at least the same level of accuracy, otherwise you may compare apples with oranges.

**Comment by guolinke** [comment]
I think this will not affect so much in accuracy. 
And it will not affect speed as well, the gain calculation cost is very small in LightGBM as well. 
However, we will set min_child_weight=1 in LightGBM and run them again. 
Thanks for your advise.

**Comment by sh1ng** [comment]
My proposal is set min_child_weight = 100 or 200 for xgboost. 
min_child_weight=1 doesn't make a lot of sense IMHO

**Comment by guolinke** [comment]
OK, LightGBM is set ```min_data_in_leaf=100```  and ```min_sum_hessian_in_leaf=10.0``` by default.
So I should set  ```min_data_in_leaf=0``` and ```min_sum_hessian_in_leaf=100.0``` in LightGBM, and set ```min_child_weight=100.0`` in xgboost. 
This should give a fair result.

**Comment by guolinke** [comment]
https://github.com/Microsoft/LightGBM/wiki/Experiments  had updated.

**Comment by RAMitchell** [comment]
@guolinke would you consider adding your scripts used to generate these experiments to the repo? 

In particular any preprocessing of the raw data files. I would like to exactly replicate these benchmarks for GPU optimisation work on xgboost.

**Comment by guolinke** [comment]
@RAMitchell I will build a new repo for the benchmarks. 
BTW, are you interesting about porting GPU accelerate to LightGBM?

**Comment by RAMitchell** [comment]
Great, thanks.

Definitely interested in implementing a version of your algorithm, although probably in xgboost as I have already done a fair bit of work there.

I think your algorithm will have a lot of nice properties for GPU parallelism.

**Comment by sh1ng** [comment]
@RAMitchell it's better to use independent benchmark like https://github.com/szilard/benchm-ml it already contains xgboost as first class citizen. You can create PR for xgboost GPU plugin.

It worth to play with regression task also, so If you know big enough dataset (10M at least) let me know.
